tasks/libs/common/gitlab.py

```diff
@@ -89,30 +89,39 @@ def last_pipeline_for_ref(self, project_name, ref, per_page=100):
 
         return sorted(pipelines, key=lambda pipeline: pipeline['created_at'], reverse=True)[0]
 
+    def trigger_pipeline(self, project_name, data):
+        """
+        Trigger a pipeline on a project using the trigger endpoint.
+        Requires a trigger token in the data object.
```

Comment:
Can you document how this should happen? (i.e. by passing the `token` field). Also, maybe we should check for it here and raise an exception if missing?